pageserver/src/tenant/timeline.rs

```diff
@@ -3652,7 +3653,10 @@ impl Timeline {
         let _write_guard = if write_lock_held {
             None
         } else {
-            Some(self.write_lock.lock().await)
+            let mut g = self.write_lock.lock().await;
+            // remove the reference to an open layer
+            g.take();
```

Comment:
I'd rather not since the follow-up refactoring clears that up? At least moves it to happen only in a single place... Well, it doesn't hurt, I'll find a place.

Reply:
Feel free to skip, but I think it would still be useful for future refactors even if with the follow up pr. Don't feel strongly at all though.

Reply:
5db078092